stubs/urllib3/urllib3/connectionpool.pyi

```diff
@@ -29,48 +35,54 @@ Retry = retry.Retry
 Timeout = timeout.Timeout
 get_host = url.get_host
 
+_Timeout: TypeAlias = Timeout | float
+_Retries: TypeAlias = Retry | bool | int
+
 xrange: Any
-log: Any
+log: Logger
 
 class ConnectionPool:
-    scheme: Any
-    QueueCls: Any
-    host: Any
-    port: Any
-    def __init__(self, host, port=...) -> None: ...
-    def __enter__(self): ...
-    def __exit__(self, exc_type, exc_val, exc_tb): ...
-    def close(self): ...
+    scheme: ClassVar[str | None]
+    QueueCls: ClassVar[type[LifoQueue]]
```

Comment:
I still think this would be better as the stdlib `Queue[Any]`. It's true that at runtime, `ConnectionPool.QueueCls` is `urllib3.util.queue.LifoQueue`. But `ConnectionPool` is a base class for other `ConnectionPool` classes, and I don't think (from my impression, from reading the code) that it's mandatory for all subclasses to have their `QueueCls` attribute be a subclass of `urllib3.util.queue.LifoQueue`. It seems to me that anything with an interface similar to the stdlib `queue.Queue` class will do. There's evidence for this in the fact that they scrapped the `urllib3.util.queue` submodule in the `main` branch and switched to using the stdlib `queue` module.

Reply:
Anyway, this is minor, other than this it looks good to me now :)

Reply:
👍  I think I got mixed up between the `type[_QueueT]` change and the `Queue[Any]` change. I agree with you on that.

If you are comfortable with it, I will update this to: `ClassVar[type[Queue[Any]]]`

and change the imports to be fully qualified to differentiate between the two different queues.

Sound good?

Reply:
That sounds good to me! 👍

Reply:
That was changed in 3e231f4 (I expect a pre-commit will trigger). The only thing I may have missed was if there was a better way to import the `LifoQueue` to avoid the overlapping module names. As it stands, I've updated it to `import queue` for the standard library instance and usage as `queue.Queue[Any]` and have left the urllib3 `LifoQueue` as `from .util.queue import LifoQueue`.

Hopefully thats correct

Reply:
How about `from .util import queue as urllib3queue`? And then refer to it as `urllib3queue.LifoQueue`

Reply:
I had thought that was going to trigger a stubtest warning about it not existing at runtime, but I could just be getting confused about what is checked and what isn't. 

That's a good approach, thank you for the suggestion, worst case we can do it as `from .util import queue as _urllib3queue` to avoid the check

Reply:
All imports in stub files are considered "private to the stub" (not re-exported), _unless_ they fall into one of the two categories:

- `import foo as foo` -- the `as` has to be truly redundant; if you do `import foo as bar`, it's still considered private, but `import foo as foo` is considered a public re-export.
- `from foo import *`

Reply:
-- A third exception is if a name is included in `__all__`, it's always considered re-exported, even if it's imported via a means that would usually make it a "private" import.